cmd/osmosisd/cmd/airdrop.go

```diff
@@ -0,0 +1,304 @@
+		Short: "Import balances from provided genesis to {FlagHome}/genesis.json",
+		Long: `Import balances from provided genesis to {FlagHome}/genesis.json
+Download:
+	https://raw.githubusercontent.com/cephalopodequipment/cosmoshub-3/master/genesis.json
+Init genesis file:
+	osmosisd init mynode
+Example:
+	osmosisd export-airdrop-genesis uatom ../genesis.json 100000000000000 --snapshot-output="../snapshot.json"
+	- Check genesis:
+		file is at ~/.osmosisd/config/genesis.json
+	- Snapshot
+		file is at "../snapshot.json"
+`,
+		Args: cobra.ExactArgs(3),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx := client.GetClientContextFromCmd(cmd)
+			depCdc := clientCtx.JSONMarshaler
+			cdc := depCdc.(codec.Marshaler)
+			aminoCodec := clientCtx.LegacyAmino.Amino
+
+			serverCtx := server.GetServerContextFromCmd(cmd)
+			config := serverCtx.Config
+
+			config.SetRoot(clientCtx.HomeDir)
+
+			denom := args[0]
+			filepath := args[1]
+			osdenom := "uosmo"
+			snapshotOutput, err := cmd.Flags().GetString(flagSnapshotOutput)
+			if err != nil {
+				return fmt.Errorf("failed to get snapshot directory: %w", err)
+			}
+
+			totalAmount, ok := sdk.NewIntFromString(args[2])
+			if !ok {
+				return fmt.Errorf("failed to parse totalAmount: %s", args[2])
+			}
+
+			genFile := config.GenesisFile()
+			appState, genDoc, err := genutiltypes.GenesisStateFromGenFile(genFile)
+			if err != nil {
+				return fmt.Errorf("failed to unmarshal genesis state: %w", err)
+			}
+
+			authGenState := authtypes.GetGenesisStateFromAppState(cdc, appState)
+
+			accs, err := authtypes.UnpackAccounts(authGenState.Accounts)
+			if err != nil {
+				return fmt.Errorf("failed to get accounts from any: %w", err)
+			}
+
+			jsonFile, err := os.Open(filepath)
+			if err != nil {
+				return err
+			}
+			defer jsonFile.Close()
+
+			byteValue, _ := ioutil.ReadAll(jsonFile)
+
+			var genStateV036 GenesisStateV036
+
+			setCosmosBech32Prefixes()
+			err = aminoCodec.UnmarshalJSON(byteValue, &genStateV036)
+			if err != nil {
+				return err
+			}
+
+			snapshot := []SnapshotFields{}
+			balanceIndexByAddress := make(map[string]int)
+			totalAtomBalance := sdk.NewInt(0)
+			for index, account := range genStateV036.AppState.Accounts {
+				totalAtomBalance = totalAtomBalance.Add(account.Coins.AmountOf(denom))
+				balanceIndexByAddress[account.Address.String()] = index
+				snapshot = append(snapshot, SnapshotFields{
+					AtomAddress: account.Address.String(),
+					AtomBalance: account.Coins.AmountOf(denom),
+					AtomPercent: sdk.NewDec(0),
+				})
+			}
+
+			for _, delegation := range genStateV036.AppState.Staking.Delegations {
+				address := delegation.DelegatorAddress
+				index, ok := balanceIndexByAddress[address.String()]
+				if !ok {
+					continue
+				}
+				sharesInt := delegation.Shares.RoundInt()
+				snapshot[index].AtomBalance = snapshot[index].AtomBalance.Add(sharesInt)
+				totalAtomBalance = totalAtomBalance.Add(sharesInt)
+			}
```

Comment:
We should just save the data on the amount from staked vs nonstaked in the snapshot array on the first iteration, so we don't have to reiterate over this again later.  Might as well include this data in the snapshot export as well.

So should add `AtomStakedBalance` and `AtomUnstakedBalance`

```
// SnapshotFields provide fields of snapshot per account
type SnapshotFields struct {
	AtomAddress string  `json:"atom_address"`
	AtomBalance sdk.Int `json:"atom_balance"`
        AtomStakedBalance sdk.Int `json:"atom_staked_balance"`
        AtomUnstakedBalance sdk.Int `json:"atom_unstaked_balance"`
	AtomPercent sdk.Dec `json:"atom_ownership_percentage"`
	OsmoAddress string  `json:"osmo_address"`
	OsmoBalance sdk.Int `json:"osmo_balance"`
	OsmoPercent sdk.Dec `json:"osmo_ownership_percentage"`
}
```